fsw/c-blinky/main.c

```diff
@@ -0,0 +1,238 @@
+// Base addresses
+#define RCC_BASE     0x58024400UL
+#define PWR_BASE     0x58024800UL
+#define GPIOE_BASE   0x58021000UL
+#define FLASH_BASE   0x52002000UL
+#define SYSTICK_BASE 0xE000E010UL
+
+// RCC registers
+#define RCC_CR       (*(volatile U32*) (RCC_BASE + 0x00))
+#define RCC_CFGR     (*(volatile U32*) (RCC_BASE + 0x10))
+#define RCC_D1CFGR   (*(volatile U32*) (RCC_BASE + 0x18))
+#define RCC_D2CFGR   (*(volatile U32*) (RCC_BASE + 0x1C))
+#define RCC_D3CFGR   (*(volatile U32*) (RCC_BASE + 0x20))
+#define RCC_PLL1CFGR (*(volatile U32*) (RCC_BASE + 0x28))
+#define RCC_PLL1DIVR (*(volatile U32*) (RCC_BASE + 0x30))
+#define RCC_AHB4ENR  (*(volatile U32*) (RCC_BASE + 0xE0))
+
+// PWR registers
+#define PWR_CSR1 (*(volatile U32*) (PWR_BASE + 0x04))
+#define PWR_CR3  (*(volatile U32*) (PWR_BASE + 0x0C))
+#define PWR_D3CR (*(volatile U32*) (PWR_BASE + 0x18))
+
+// GPIO registers
+#define GPIOE_MODER   (*(volatile U32*) (GPIOE_BASE + 0x00))
+#define GPIOE_OTYPER  (*(volatile U32*) (GPIOE_BASE + 0x04))
+#define GPIOE_OSPEEDR (*(volatile U32*) (GPIOE_BASE + 0x08))
+#define GPIOE_ODR     (*(volatile U32*) (GPIOE_BASE + 0x14))
+
+// Other registers
+#define FLASH_ACR   (*(volatile U32*) (FLASH_BASE + 0x00))
+#define SYSTICK_CSR (*(volatile U32*) (SYSTICK_BASE + 0x00))
+#define SYSTICK_RVR (*(volatile U32*) (SYSTICK_BASE + 0x04))
+#define SYSTICK_CVR (*(volatile U32*) (SYSTICK_BASE + 0x08))
+
+// Bit definitions
+#define RCC_CR_HSEON             (1 << 16)
+#define RCC_CR_HSERDY            (1 << 17)
+#define RCC_CR_HSEBYP            (1 << 18)
+#define RCC_CR_PLL1ON            (1 << 24)
+#define RCC_CR_PLL1RDY           (1 << 25)
+#define RCC_CFGR_SW_PLL1         (0x3 << 0)
+#define RCC_CFGR_SWS_PLL1        (0x3 << 3)
+#define RCC_D1CFGR_D1CPRE_DIV1   (0x0 << 8)
+#define RCC_D1CFGR_HPRE_DIV2     (0x8 << 0)
+#define RCC_D2CFGR_D2PPRE1_DIV2  (0x4 << 4)
+#define RCC_D2CFGR_D2PPRE2_DIV2  (0x4 << 8)
+#define RCC_D3CFGR_D3PPRE_DIV2   (0x4 << 4)
+#define RCC_PLL1CFGR_PLL1SRC_HSE (0x2 << 0)
+#define RCC_PLL1CFGR_PLL1PEN     (1 << 16)
+#define RCC_PLL1CFGR_PLL1QEN     (1 << 17)
+#define RCC_AHB4ENR_GPIOEEN      (1 << 4)
+#define PWR_CR3_LDOEN            (1 << 1)
+#define PWR_CR3_BYPASS           (1 << 0)
+#define PWR_CR3_SDEN             (1 << 2)
+#define PWR_D3CR_VOS_SCALE1      (0x3 << 14)
+#define PWR_CSR1_ACTVOSRDY       (1 << 13)
+#define FLASH_ACR_LATENCY_4WS    (4 << 0)
+#define FLASH_ACR_PRFTEN         (1 << 8)
+#define FLASH_ACR_ICEN           (1 << 9)
+#define FLASH_ACR_DCEN           (1 << 10)
+#define SYSTICK_CSR_ENABLE       (1 << 0)
+#define SYSTICK_CSR_TICKINT      (1 << 1)
+#define SYSTICK_CSR_CLKSOURCE    (1 << 2)
+
+// Constants
+#define RED_LED_PIN        5
+#define PLL1_M             3
+#define PLL1_N             100
+#define PLL1_P             2
+#define PLL1_Q             8
+#define RTT_BUFFER_SIZE_UP 1024
+
+// RTT structures
+typedef struct {
+    const char*       sName;
+    char*             pBuffer;
+    unsigned          SizeOfBuffer;
```

Comment:
This might be a C style thing I'm not familiar with, but some of these field names are in PascalCase and others are in camelCase